### ISSUE TITLE
User configurable mouse bindings (like keymap for key+mouse)

### ISSUE THREAD

**Issue by JosephTLyons** [zed-industries/zed#10647]
Support for user-defined mappings of (for example):

- `mouse1` - left click
- `mouse2` - right click
- `mouse3` - toggle scroll, paste, jump to definition
- `mouse3 drag` - multi-cursor
- `alt-mouse1` - open url (match vscode linux)
- `alt-mouse1` - columnular selection (see: [`COLUMNAR_SELECTION_MODIFIERS`](https://github.com/zed-industries/zed/blob/fc262feb3ccdb7cfb987b83e75f0eae8b726adb9/crates/editor/src/editor.rs#L246-L252))
- `alt-shift-mouse1` - multi-cursor
- `shift-mouse1` - close tab
- `ctrl-mouse1` - right click (macos default)
- `ctrl-mousescrolldown` - Font size increase 
- `ctrl-mousescrollup` - Font size decrease
- `alt-mousescrolldown` - Fast scroll down
- `alt-mousescrollup` - Fast scroll up
- `force-click` - GotoDefinition (Jetbrains default)
etc.

Related issues:
- https://github.com/zed-industries/zed/issues/13295
- https://github.com/zed-industries/zed/issues/11990
- https://github.com/zed-industries/zed/issues/11786
- https://github.com/zed-industries/zed/issues/9745
- https://github.com/zed-industries/zed/issues/14612
- https://github.com/zed-industries/zed/issues/4644

**Comment by micahscopes** [comment]
Middle mouse click-to-paste is making me feel very disoriented with zed on my linux laptop.  I can't find a way to disable.  If I accidentally touch three fingers to the touchpad as I'm scrolling I'll sprinkle little random pastes everywhere.

**Comment by micahscopes** [comment]
In the meantime I made a PR for disabling linux middle mouse paste: https://github.com/zed-industries/zed/pull/16572

**Comment by forresthopkinsa** [comment]
I am so used to middle-clicking to go to definition / go to reference in other IDEs, this really throws off my muscle memory when switching to Zed for tasks

**Comment by RustoMCSpit** [comment]
mouse4 and mouse5 too

**Comment by lukkoro** [comment]
If you have a Logitech mouse, you can set it up to mimic keyboard shortcut on click, which is what I did, and it works perfectly

**Comment by tracure1337** [comment]
Please please give us mouse settings in Zed. Muy importante :-)

**Comment by ferdimsu** [comment]
Are there any updates on this, since a lot of people use mouse scrolling to increase or decrease their font size (font zooming)? Most people are also accustomed to using Ctrl + scroll for font size adjustment.

**Comment by pjv** [comment]
Looking very forward to being able to change the mouse binding for columnar selection from shift-alt (option) to just alt (option). Muscle memory.

**Comment by itsaphel** [comment]
Presumably this would also enable https://github.com/zed-industries/zed/issues/5045?

**Comment by ivanstepanovftw** [comment]
This is not high priority now?
https://github.com/zed-industries/zed/issues?q=state%3Aopen%20label%3A%22priority%22%20-label%3Abug%20sort%3Areactions-%2B1-desc
It has the same amount of likes as #6722

**Comment by ivanstepanovftw** [comment]
I do not have second spare hand to press Ctrl to go to definition.
I do not want to press Ctrl and tap at the same time with 1 hand. Try it by yourself.
I do not want to tap and move my hand to F12.
I do not want to use Windows with AutoHotKey.
I do not want to use X11 instead of Wayland to script middle mouse click to button press and F12 key press.

I want to use 1 hand and 3 fingers tap (middle click) to go to definition.

**Comment by ivanstepanovftw** [comment]
Remapping does not work. I have tried Input Remapper (no touchpad support for me).
Not a solution to VSCode and Zed:
```
modify(Control_L, key(BTN_LEFT))
```
Solution to VSCode, but not for Zed:
```
hold_keys(Control_L, BTN_LEFT)
```
Solution to Zed and VSCode:
```
key_down(Control_L).wait(10).key(BTN_LEFT).wait(100).key_up(Control_L)
```

I think Rust input stack does not know what key modifier is and listens to keypresses or something like that

<img width="2104" height="1486" alt="Image" src="https://github.com/user-attachments/assets/f9730168-1779-43bc-ac53-24e1ebff9902" />

**Comment by makinori** [comment]
Not being able to use my scroll wheel and other mouse bindings is the reason I haven't been able to use Zed yet

**Comment by danmou** [comment]
Still relevant.

**Comment by miguelraz** [comment]
Sorry for the noise, the `stalebot` upstream went haywire.

Please disregard it.

**Comment by ARYPROGRAMMER** [comment]
hey, @as-cii , @maxbrunsfeld and other maintainers. I think this issue should be prioritized. I am unable to configure ctrl + mouse wheel to zoom in and out in the code file which i think should be fixed. Since there's no one picking this up right now I think you guys can assign this issue to me, I think i'll do it in the way its useful for all as this might be a real pain in the neck. 

In the meantime if there's a way around to temporary fix this i'll appreciate if someone finds it or even i do.

thanks, cc: @JosephTLyons

**Comment by BuruY** [comment]
+1 for `super+click`: Go To Definition / Reference
